### ISSUE TITLE
MA0076 does not trigger for IEnumerable.Count()

### ISSUE THREAD

**Issue by 0liver** [meziantou/Meziantou.Analyzer#291]
**Context**
Rule identifier: _MA0076_
Target Framework: _.NET Fx 4.6_
Language version: _7.3_

**Describe the bug**
The rule does not trigger a warning when interpolating `IEnumerable.Count()`.

**To Reproduce**
Code or steps to reproduce the behavior:

````c#
var enumerable = Enumerable.Empty<int>();
// Does not trigger MA0076:
var stringified = $"{enumerable.Count()}";

var count = enumerable.Count();
// Does trigger MA0076:
var stringified2 = $"{count}";
````

Maybe this is by design, but I was surprised to see the warning appear when I extracted an `IEnumerable.Count()` statement into a variable.

**Comment by meziantou** [comment]
It's by design. `Count()` is always a positive number. Positive numbers are not sensible to the current culture ([source](https://source.dot.net/#System.Private.CoreLib/Int32.cs,90))

The analyzer doesn't consider the data flow. So, if you extract a variable, it cannot detect the value of the variable is a positive number at the usage site.

**Comment by 0liver** [comment]
Thanks a lot! I wasn't aware of that! Basically all numbers in our source code are positive - either DB row IDs, or counts. Should we be using `uint` to avoid having to call `.ToString(CultureInfo.InvariantCulture)` everywhere?

**Comment by meziantou** [comment]
I think you should disable the rule (`.editorconfig`) if you think it is not an issue in your codebase

**Comment by 0liver** [comment]
Thanks for your time and effort.
